src/compiler/diagnosticMessages.json

```diff
@@ -3523,6 +3523,14 @@
         "category": "Error",
         "code": 8024
     },
+    "Generic type '{0}' requires '{1}' type arguments; provide these with an '@extends' tag.": {
+        "category": "Error",
+        "code": 8025
+    },
+    "Generic type '{0}' requires between '{1}' and '{2}' type arguments; provide these with an '@extends' tag.": {

```

Comment:
Basing it on the new error message is better: "Expected 1-2 type arguments, but got 3; provide them with an '@extends' tag."

I didn't notice when I added the new error message that there was an existing one. I filed #19374 to fix that.